sklearn/naive_bayes.py

```diff
@@ -1339,14 +1340,14 @@ def _more_tags(self):
     def _check_X(self, X):
         """Validate X, used only in predict* methods."""
         X = self._validate_data(
-            X, dtype="int", accept_sparse=False, force_all_finite=True, reset=False
+            X, dtype="int", accept_sparse="csr", force_all_finite=True, reset=False
```

Comment:
Given how `X` is accessed column-wise (with `X[:, i]`), I think this is better as CSC.

Reply:
Thank you for the review. I'm with you. Probably, How about changing `accept_sparse=True` and converting a given matrix to csc if it is not csc.

Reply:
By the way, even if you specify `csr`, CSC matrix can be accepeted. Is it a bug?

Reply:
If `accept_sparse='csc'` and a CSR matrix is passed in, `check_array` will convert:

https://github.com/scikit-learn/scikit-learn/blob/40f94605a745f1c4b50284689210850e31d4881c/sklearn/utils/validation.py#L645-L646

Reply:
I missed it. Thank you.

Reply:
So what about considering `accept_sparse="csc"` as initially suggested by Thomas above?

Reply:
Yes, it sounds good. I changed it.